### ISSUE TITLE
'Declare and assign separately to avoid masking return values. [SC2155]' isn't triggered for declare/local with arguments

### ISSUE THREAD

**Issue by andrey-starodubtsev** [koalaman/shellcheck#558]
```bash
    #!/bin/bash

    a="a  b"
    declare b=$(echo "$a" | grep e) # has SC2155 warning

    declare -a c=($(echo "$a" | grep c)) # has no SC2155 warning
    echo "$?" # prints 0

    declare -a d
    d=($(echo "$a" | grep c))
    echo "$?" # prints 1

    declare -r e=($(echo "$a" | grep c)) # has no SC2155 warning
    echo "$?" # prints 0

    f=($(echo "$a" | grep c))
    echo "$?" # prints 1
    declare -r f
    f=("$f{@}" 1) # ./1.sh: line 22: f: readonly variable

    export b c d e # supress shellcheck's warning about unused variables
```

I've read note about readonly variables in https://github.com/koalaman/shellcheck/wiki/SC2155, but it's possible to declare variable as readonly (right) _after_ its assignment, so I think that it would be useful to be able to find `readonly`/`local -r`/`declare -r` coupled with assignments, at least optionally.
And anyway, bash arrays are still an issue.

**Comment by timoreimann** [comment]
Though a rather old issue, I find this quite important, especially with regards to `readonly`. Putting `readonly`, assignments, and command substitutions on the same line, i.e.,

`readonly foo=$(this_could_fail)`

is a common mistake easily and perfectly solved by breaking things down into two lines:

```
foo=$(this_could_fail)
readonly foo
```

I see no reason why this shouldn't be marked as SC2155 (or a new rule).

**Comment by WalterS** [comment]
:( Habe ich natürlich lokal getestet, aber die von Red Hat ausgelieferte Version (0.3.5) meckert da nicht.

**Comment by kurahaupo** [comment]
The problem is that to replace `local -r foo=$( cmd )` you need _three_ lines:
```
local var
var=$( cmd )
readonly var
